### ISSUE TITLE
Handle preprocessor calls used as loops

### ISSUE THREAD

**Issue by Elscha** [KernelHaven/srcMLExtractor#8]
The following code cannot be parsed:
```
for_each_possible_cpu(cpu) {
    xen_get_runstate_snapshot_cpu_delta(&state, cpu);
        memcpy(runstate_delta[cpu].time, state.time,
            sizeof(runstate_delta[cpu].time));
}
```
This is part of the robustness tests. However, you need to rename `time.c_deactivated`into `time.c`

**Comment by adam-sse** [comment]
This is not an anonymous function, but a preprocessor call:
```C
// include/linux/cpumask.h
#if NR_CPUS == 1

#define for_each_cpu(cpu, mask)                 \
        for ((cpu) = 0; (cpu) < 1; (cpu)++, (void)mask)

#else

#define for_each_cpu(cpu, mask)                         \
        for ((cpu) = -1;                                \
                (cpu) = cpumask_next((cpu), (mask)),    \
                (cpu) < nr_cpu_ids;)

#endif

#define for_each_possible_cpu(cpu) for_each_cpu((cpu), cpu_possible_mask)

// time.c
for_each_possible_cpu(cpu) {
	xen_get_runstate_snapshot_cpu_delta(&state, cpu);
	memcpy(runstate_delta[cpu].time, state.time,
			sizeof(runstate_delta[cpu].time));
}
```

**Comment by adam-sse** [comment]
Fixed by b301296e4dcbc1a11ef612ae62ae743a07a34d27.

This is now translated into:
```
[1] Statement:
	[1] for_each_possible_cpu ( cpu )
[1] CompoundStatement
	[1] Statement:
		[1] xen_get_runstate_snapshot_cpu_delta ( & state , cpu ) ;
	[1] Statement:
		[1] memcpy ( runstate_delta [ cpu ] . time , state . time , sizeof ( runstate_delta [ cpu ] . time ) ) ;

```

A similar issue in `testdata/real/Linux4.15/capability.c` was fixed by this, too:
```C
// include/linux/syscalls.h
#define SYSCALL_DEFINE2(name, ...) SYSCALL_DEFINEx(2, _##name, __VA_ARGS__)

#define SYSCALL_DEFINEx(x, sname, ...)                          \
        SYSCALL_METADATA(sname, x, __VA_ARGS__)                 \
        __SYSCALL_DEFINEx(x, sname, __VA_ARGS__)

#define __PROTECT(...) asmlinkage_protect(__VA_ARGS__)
#define __SYSCALL_DEFINEx(x, name, ...)                                 \
        asmlinkage long sys##name(__MAP(x,__SC_DECL,__VA_ARGS__))       \
                __attribute__((alias(__stringify(SyS##name))));         \
        static inline long SYSC##name(__MAP(x,__SC_DECL,__VA_ARGS__));  \
        asmlinkage long SyS##name(__MAP(x,__SC_LONG,__VA_ARGS__));      \
        asmlinkage long SyS##name(__MAP(x,__SC_LONG,__VA_ARGS__))       \
        {                                                               \
                long ret = SYSC##name(__MAP(x,__SC_CAST,__VA_ARGS__));  \
                __MAP(x,__SC_TEST,__VA_ARGS__);                         \
                __PROTECT(x, ret,__MAP(x,__SC_ARGS,__VA_ARGS__));       \
                return ret;                                             \
        }                                                               \
        static inline long SYSC##name(__MAP(x,__SC_DECL,__VA_ARGS__))

// capability.c
SYSCALL_DEFINE2(capget, cap_user_header_t, header, cap_user_data_t, dataptr)
{
	/* ... */
}
```

This is now translated into:
```
[1] Statement:
	[1] SYSCALL_DEFINE2 ( capget , cap_user_header_t , header , cap_user_data_t , dataptr )
[1] CompoundStatement
	[1] Comment:
		[1] /* ... */
```
Sadly, we cannot detect that this is supposed to be a function.
